[docs] Document branch-delete workflow job (#46914)
expo/expo: docs/pages/eas/workflows/examples/branch-cleanup.mdx

---
title: Clean up EAS Update branches with EAS Workflows
sidebar_title: Clean up update branches

> description: Learn how to delete EAS Update branches when GitHub branches are deleted.

Reviewer: ```suggestion
description: Learn how to delete EAS Update branches when GitHub branches are deleted using EAS Workflows.
```

Missing context.

> hideTOC: true

Reviewer: ```suggestion
```

I don't think we should hide it. There are 3 sections on this page so I think it is okay to show the TOC.

---

import { Prerequisites, Requirement } from '~/ui/components/Prerequisites';
import { Terminal } from '~/ui/components/Snippet';

> If you publish updates to EAS Update branches named after GitHub branches, for example with [`eas update --auto`](/eas-update/eas-cli/#create-a-new-update-and-publish-it), the [`update` workflow job](/eas/workflows/pre-packaged-jobs#update) with `branch: ${{ github.ref_name }}`, or [preview updates on every branch](/eas/workflows/examples/publish-preview-update), deleting the GitHub branch leaves the EAS Update branch behind. This workflow cleans up when a Git branch is deleted.

Reviewer: **[suggestion]** This opening sentence runs ~45 words with three embedded links before the main clause, so the consequence lands late (`style-guide.md#voice-and-tone`). Split the list of options from the result:

```suggestion
You might publish updates to EAS Update branches named after GitHub branches, for example with [`eas update --auto`](/eas-update/eas-cli/#create-a-new-update-and-publish-it), the [`update` workflow job](/eas/workflows/pre-packaged-jobs#update) with `branch: ${{ github.ref_name }}`, or [preview updates on every branch](/eas/workflows/examples/publish-preview-update). When you do, deleting the GitHub branch leaves the EAS Update branch behind. This workflow cleans up the orphaned EAS Update branch when a Git branch is deleted.
```

    <Terminal cmd={['$ eas update:configure']} />

  </Requirement>
  <Requirement title="Connect your GitHub repository">
    Your EAS project must be linked to a GitHub repository. See [Get started with EAS Workflows](/eas/workflows/get-started/#automate-workflows-with-github-events).
  </Requirement>
  <Requirement title="Workflow file on the default branch">
    Keep this workflow file on your repository's default branch. When a branch is deleted, EAS reads workflow configuration from the default branch HEAD, not from the deleted ref.
  </Requirement>
</Prerequisites>

The following workflow deletes the EAS Update branch with the same name as the deleted GitHub branch:

```yaml .eas/workflows/branch-cleanup.yml
name: Branch cleanup

on:
  ref_delete:
    branches: ['*', '!main']

jobs:
  delete-update-branch:
    type: branch-delete
    params:
      branch_name: ${{ github.ref_name }}
```

## How it works

1. [`on.ref_delete`](/eas/workflows/syntax/#onref_delete) runs when a GitHub branch matching the patterns is deleted. The example above triggers on all branches except `main`.

> 2. The [`branch-delete`](/eas/workflows/pre-packaged-jobs#branch-delete) job deletes the EAS Update branch named `${{ github.ref_name }}` — the same name as the deleted Git branch.

Reviewer: **[nit]** Avoid em dashes; a comma carries the appositive (`style-guide.md#punctuation`).

```suggestion
2. The [`branch-delete`](/eas/workflows/pre-packaged-jobs#branch-delete) job deletes the EAS Update branch named `${{ github.ref_name }}`, the same name as the deleted Git branch.
```
